auslib/db.py

```diff
@@ -1539,18 +1539,28 @@ def _buildIDMatchesRule(self, ruleBuildID, queryBuildID):
             return True
         return string_compare(queryBuildID, ruleBuildID)
 
-    def _csvMatchesRule(self, ruleString, queryString, substring=True):
-        """Decides whether a column from a rule matches an incoming one.
-           Some columns in a rule may specify multiple values delimited by a
-           comma. Once split we do a full or substring match against the query
-           string. Because we support substring matches, there's no need
-           to support globbing as well."""
+    def _simpleBooleanMatchesSubRule(self, subRuleString, queryStringList):
+        for rule in subRuleString:
+            if rule not in queryStringList:
+                return False
+        return True
+
+    def _simpleBooleanMatchesRule(self, ruleString, queryString):
+        """Decides whether a column from a rule matches an incoming one using simplified boolean logic.
+        Only two operators are supported: '&&' (and), ',' (or). A rule like 'AMD,SSE' will match incoming
+        rules that contain either 'AMD' or 'SSE'. A rule like 'AMD&&SSE' will only match incoming rules
+        that contain both 'AMD' and 'SSE'.
+        Rules are matched to full strings, so if we have a rule that specifies 'AMD,SSE', it will NOT be
```

Comment:
Ah, thanks for clarifying that! I thought we were moving to exact string matches across the board, but I was wrong. I added back the ability to do substring matches - OS_VERSION does this, whereas SYSTEM_REQUIREMENTS does full string matching.